openvdb_ax/openvdb_ax/compiler/LeafLocalData.h

```diff
@@ -0,0 +1,229 @@
+// Copyright Contributors to the OpenVDB Project
+// SPDX-License-Identifier: MPL-2.0
+
+/// @file compiler/LeafLocalData.h
+///
+/// @authors Nick Avramoussis
+///
+/// @brief  Thread/Leaf local data used during execution over OpenVDB Points
+///
+
+#ifndef OPENVDB_AX_COMPILER_LEAF_LOCAL_DATA_HAS_BEEN_INCLUDED
+#define OPENVDB_AX_COMPILER_LEAF_LOCAL_DATA_HAS_BEEN_INCLUDED
+
+#include <openvdb/openvdb.h>
+#include <openvdb/points/AttributeArray.h>
+#include <openvdb/points/PointAttribute.h>
+#include <openvdb/points/PointDataGrid.h>
+#include <openvdb/points/PointGroup.h>
+
+namespace openvdb {
+OPENVDB_USE_VERSION_NAMESPACE
+namespace OPENVDB_VERSION_NAME {
+
+namespace ax {
+namespace compiler {
+
+
+/// @brief  Various functions can request the use and initialization of point data from within
+///         the kernel that does not use the standard attribute handle methods. This data can
+///         then be accessed after execution to perform post-processes such as adding new groups,
+///         adding new string attributes or updating positions.
+///
+/// @note  Due to the way string handles work, string write attribute handles cannot
+///        be constructed in parallel, nor can read handles retrieve values in parallel
+///        if there is a chance the shared metadata is being written to (with set()).
+///        As the compiler allows for any arbitrary string setting/getting, leaf local
+///        maps are used for temporary storage per point. The maps use the string array
+///        pointers as a key for later synchronization.
+///
+struct LeafLocalData
```

Comment:
If this is just point-related, might not be a bad idea to include Point in the name of the class and the header?

Reply:
Done and moved to codegen/